sdk/ai/ai-projects/samples-dev/agents/sample_agents_bing_grounding.ts

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * 
+ * FILE: sample_agents_bing_grounding.ts
+ *
+ * DESCRIPTION:
+ *  This sample demonstrates how to use agent operations with the Grounding with Bing Search tool from
+ *  the Azure Agents service using a asynchronous client.
+ *
+ * USAGE:
+ *  npx ts-node sample_agents_bing_grounding.ts
+ *
+ *  Before running the sample:
+ *
+ *  npm install @azure/ai-projects @azure/identity @azure/core-util dotenv
+ *
+ *  Set this environment variables with your own values:
+ *  AZURE_AI_PROJECTS_CONNECTION_STRING - the Azure AI Project connection string, as found in your AI Studio Project
+ *  BING_CONNECTION_NAME - the name of the connection with Bing search grounding
+ */
+
+import { AIProjectsClient, fromConnectionIds, connectionToolType } from "@azure/ai-projects"
+import { delay } from "@azure/core-util";
+import { DefaultAzureCredential } from "@azure/identity";
+
+import * as dotenv from "dotenv";
+dotenv.config();
+
+const connectionString = process.env["AZURE_AI_PROJECTS_CONNECTION_STRING"] || "<endpoint>>;<subscription>;<resource group>;<project>";
+
+export async function main(): Promise<void> {
+  // Create an Azure AI Client from a connection string, copied from your AI Studio project.
+  // At the moment, it should be in the format "<HostName>;<AzureSubscriptionId>;<ResourceGroup>;<HubName>"
+  // Customer needs to login to Azure subscription via Azure CLI and set the environment variables
+  const client = AIProjectsClient.fromConnectionString(connectionString || "", new DefaultAzureCredential());
+  const bingConnection = await client.connections.getConnection(process.env["BING_CONNECTION_NAME"] || "<connection-name>");
+  const connectionId = bingConnection.id;
+
+  // Initialize agent bing tool with the connection id
+  const bingTool = fromConnectionIds(connectionToolType.BingGrounding, [connectionId]);
+
+  // Create agent with the bing tool and process assistant run
+  const agent  = await client.agents.createAgent(
+    "gpt-4-0125-preview", {
+      name: "my-agent", 
+      instructions: "You are a helpful agent",
+      tools: [bingTool]
+    }, {
+      headers: {"x-ms-enable-preview": "true"}
```

Comment:
Is it still needed.

Reply:
I'm not sure - I tested it several times with and without the header and it's hard to tell if it's working correctly given that the results vary and the e2e is relatively unreliable (doesn't work with or without headers every few attempts). The python sample has it and the c# one doesn't. Should I remove it?

Reply:
Confirmed this header is in both c# and python samples